src/Sulu/Component/Content/Document/Subscriber/RouteSubscriber.php

```diff
@@ -38,10 +72,62 @@ public function handleMetadataLoad(MetadataLoadEvent $event)
             return;
         }
 
-        $metadata->addFieldMapping('targetDocument', [
-            'encoding' => 'system',
-            'property' => self::DOCUMENT_TARGET_FIELD,
-            'type' => 'reference',
-        ]);
+        $metadata->addFieldMapping(
+            'created',
+            [
+                'encoding' => 'system',
+                'property' => self::CREATED,
+                'type' => 'date',
+            ]
+        );
+    }
+
+    public function handlePersist(PersistEvent $event)
+    {
+        $document = $event->getDocument();
+
+        if (!$document instanceof RouteBehavior) {
+            return;
+        }
+
+        $node = $event->getNode();
+        $session = $node->getSession();
+
+        /** @var PageDocument $targetDocument */
+        $targetDocument = $document->getTargetDocument();
+        $targetNode = $this->documentInspector->getNode($targetDocument);
+
+        // copy new route to old position
+        $webspaceKey = $targetDocument->getWebspaceName();
+        $locale = $targetDocument->getLocale();
+
+        $routePath = $this->sessionManager->getRoutePath($webspaceKey, $locale, null)
+            . $targetDocument->getResourceSegment();
```

Comment:
We are assuming the document has a ResourceSegment .. do we ensure that this is an instance of `ResourceSegmentBehavior`? If not I would either create a method with the logic which accepts an object of that type or add a manual type check and throw an exception.